docs/mkdocs/documentation/deploy_kmod.md

```diff
@@ -0,0 +1,144 @@
+# Deploying kernel modules
+
+For each `Module`, KMM can create a number of `DaemonSets`:
+
+- one ModuleLoader DaemonSet per compatible kernel version running in the cluster;
+- one device plugin DaemonSet, if configured.
+
+### ModuleLoader
+
+The ModuleLoader DaemonSets run ModuleLoader images to load kernel modules.  
+A ModuleLoader image is an OCI image that contains the `.ko` files and both the `modprobe` and `sleep` binaries.
+
+When the ModuleLoader pod is created, it will run `modprobe` to insert the specified module into the kernel.
+It will then `sleep` until it is terminated.
+When that happens, the `ExecPreStop` hook will run `modprobe -r` to unload the kernel module.
+
+Learn more about [how to build a ModuleLoader image](module_loader_image.md).
+
+### Device plugin
+
+If `.spec.devicePlugin` is configured in a `Module`, then KMM will create a [device plugin](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/)
+DaemonSet in the cluster.
+That DaemonSet will target nodes:
+
+- that match the `Module`'s `.spec.selector`;
+- on which the kernel module is loaded (where the ModuleLoader pod has the `Ready` condition).
+
+## `Module` CRD
+
+The `Module` Custom Resource Definition represents a kernel module that should be loaded on all or select nodes in the
+cluster, through a ModuleLoader image.
+A Module specifies one or more kernel versions it is compatible, as well as a node selector.
+
+The compatible versions for a `Module` are listed under `.spec.moduleLoader.container.kernelMappings`.
+A kernel mapping can either match a `literal` version, or use `regexp` to match many of them at the same time.
+
+The reconciliation loop for `Module` runs the following steps:
+
+- list all nodes matching `.spec.selector`;
+- build a set of all kernel versions running on those nodes;
+- for each kernel version:
+  - go through `.spec.moduleLoader.container.kernelMappings` and find the appropriate container image name.
+    If the kernel mapping has `build` or `sign` defined and the container image does not already exist, run the build
+    and / or signing job as required;
+  - create a ModuleLoader `DaemonSet` with the container image determined at the previous step;
+  - if `.spec.devicePlugin` is defined, create a device plugin `DaemonSet` using the configuration specified under
+    `.spec.devicePlugin.container`;
+- garbage-collect existing `DaemonSets` targeting kernel versions that are not run by any node in the cluster.
```

Comment:
we should also add:
- garbage collect successfully finished build jobs
- garbage collect successfully finished sign jobs